File: tests/java/tests-java-gradle-junit5-allure3-selenide/src/test/java/api/ApiTestBase.java
package api;

import tests.AllureMeta;
import config.ConfigReader;
import config.TestConfig;
import io.restassured.RestAssured;
import io.restassured.specification.RequestSpecification;
import org.junit.jupiter.api.BeforeAll;

public class ApiTestBase extends AllureMeta {

    protected static final TestConfig config = ConfigReader.testConfig;

    /** Per-request spec: {@code given(jsonSpec())}. */
    protected static RequestSpecification jsonSpec() {
        return ApiSpecs.json();
    }

    @BeforeAll
    static void logFailedHttp() {
        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
    }
}

File: tests/kotlin/tests-kotlin-gradle-junit5-allure3-playwright/src/test/kotlin/api/ApiTestBase.kt
package api

import annotations.Framework
import config.ConfigReader
import config.TestConfig
import org.junit.jupiter.api.BeforeAll
import tests.AllureMeta

@Framework("ktor")
open class ApiTestBase : AllureMeta() {

    companion object {
        @JvmField
        val config: TestConfig = ConfigReader.testConfig

        @JvmStatic
        @BeforeAll
        fun setupKtor() {
            KtorHttp.setup(config)
        }
    }
}
